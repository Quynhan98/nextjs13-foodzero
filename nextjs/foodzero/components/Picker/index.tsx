import { memo, useState } from 'react'
import { Input, InputProps } from '@chakra-ui/react'
import DatePicker from 'react-datepicker'
import 'react-datepicker/dist/react-datepicker.css'

// Utils
import { startDateFrom, startDateTo } from '@utils/index'

interface DatePickerProps extends InputProps {
  isMinDate?: boolean
  isMaxDate?: boolean
  onChangeDate: (date: Date) => void
  disabled?: boolean
  pickerValue?: string
}

const Picker = ({
  isMinDate,
  isMaxDate,
  onChangeDate,
  disabled,
  pickerValue,
  ...rest
}: DatePickerProps) => {
  const [startDate, setStartDate] = useState<Date | null>()

  const handleChange = (dateTime: Date) => {
    setStartDate(dateTime)
    onChangeDate(dateTime)
  }

  return (
    <DatePicker
      selected={!pickerValue ? null : startDate}
      onChange={handleChange}
      placeholderText="MM/DD/YYYY"
      dropdownMode="scroll"
      minDate={isMinDate ? startDateFrom : null}
      maxDate={isMaxDate ? startDateTo : null}
      disabled={disabled}
      useShortMonthInDropdown
      customInput={<Input {...rest} />}
    />
  )
}

export default memo(Picker)
