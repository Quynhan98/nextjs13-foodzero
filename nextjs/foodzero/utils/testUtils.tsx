import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { BookingProvider } from '@contexts/BookingProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <AuthProvider>
        <BookingProvider>{component}</BookingProvider>
      </AuthProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
