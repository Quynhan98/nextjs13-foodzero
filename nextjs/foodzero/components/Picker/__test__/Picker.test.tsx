import { fireEvent, render, screen } from '@utils/testUtils'

// Component
import Picker from '@components/Picker'

describe('Picker Component', () => {
  const props = {
    isMinDate: true,
    isMaxDate: true,
    onChangeDate: jest.fn(),
  }

  it('renders picker component', () => {
    const { container } = render(<Picker {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('Should onchange date picker ', () => {
    render(<Picker {...props} />)

    const input = screen.getByRole('textbox')

    fireEvent.change(input, { target: { value: '12/09/2022' } })
    expect(input).toHaveValue('12/09/2022')
  })
})
