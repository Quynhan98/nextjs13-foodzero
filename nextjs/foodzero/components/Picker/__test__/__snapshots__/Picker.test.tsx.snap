// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker Component renders picker component 1`] = `
<div>
  <div
    class="react-datepicker-wrapper"
  >
    <div
      class="react-datepicker__input-container"
    >
      <input
        class="chakra-input css-1njo91c"
        placeholder="MM/DD/YYYY"
        value=""
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
